src/fcc_bridge_node_commands.cpp

```diff
@@ -0,0 +1,465 @@
+    mavsdk::Mission::MissionPlan takeoff_mission_plan;
+
+    // Insert takeoff action into mission plan
+    takeoff_mission_plan.mission_items.push_back(takeoff_mission_item);
+
+    // Execute mission
+    if (!this->execute_mission_plan(takeoff_mission_plan)) {
+        RCLCPP_FATAL(this->get_logger(),
+                     "There was an error triggering the waypoint mission! "
+                     "Triggering RTH...");
+        this->trigger_rth();
+        return;
+    }
+
+    // Set state to waiting for mission
+    this->set_internal_state(INTERNAL_STATE::FLYING_MISSION);
+
+    RCLCPP_INFO(
+        this->get_logger(),
+        "Successfully triggered takeoff. Waiting for mission to complete.");
+}
+
+void FCCBridgeNode::start_flying_to_waypoint(
+    const interfaces::msg::Waypoint &waypoint, const float speed_mps) {
+    RCLCPP_INFO(this->get_logger(),
+                "Received a command to fly to a waypoint at lat: %f°\tlon: "
+                "%f°\trel alt: %fm with speed: %fm/s",
+                waypoint.latitude_deg, waypoint.longitude_deg,
+                waypoint.relative_altitude_m, static_cast<double>(speed_mps));
+
+    // Verify that the uav is in the right state
+    switch (this->get_internal_state()) {
+        case INTERNAL_STATE::ERROR:
+            // This should never happen as the process should have exited before
+            throw std::runtime_error(
+                "Received command to take of while in ERROR state");
+        case INTERNAL_STATE::STARTING_UP:
+        case INTERNAL_STATE::ROS_SET_UP:
+        case INTERNAL_STATE::MAVSDK_SET_UP:
+        case INTERNAL_STATE::WAITING_FOR_ARM:
+        case INTERNAL_STATE::LANDED:
+        case INTERNAL_STATE::ARMED:
+            // The UAV is still or again on the ground so the only thing left is
+            // to kill the process
+            RCLCPP_FATAL(
+                this->get_logger(),
+                "Received a takeoff command in an invalid state! Exiting...");
+            this->set_internal_state(INTERNAL_STATE::ERROR);
+            this->exit_process_on_error();
+        case INTERNAL_STATE::FLYING_MISSION:
+        case INTERNAL_STATE::LANDING:
+        case INTERNAL_STATE::RETURN_TO_HOME:
+            // If the UAV is already waiting on a UAVCommand and another
+            // waypoint command is received something must have gone wrong so an
+            // RTH is triggered
+            RCLCPP_ERROR(this->get_logger(),
+                         "Received a waypoint command while there is already a "
+                         "command in progress! Triggering RTH...");
+            this->trigger_rth();
+            return;
+        case INTERNAL_STATE::WAITING_FOR_COMMAND:
+            break;
+        default:
+            throw std::runtime_error(
+                std::string("Got invalid value for internal_state: ") +
+                std::to_string(static_cast<int>(this->get_internal_state())));
+    }
+
+    // Check that the speed is valid
```

Comment:
Maybe use a helper function for this as it seems to be the same function as in lines 61-86?

Reply:
The problem is that the resulting action of an invalid speed differs, as for a takeoff command we are still on ground -> no rth.
Most of the lines are also just Log statements. So in this case I do not really see a cost effective way of putting this into its own function.